vividus-plugin-rest-api/src/main/java/org/vividus/archive/steps/ArchiveSteps.java

```diff
@@ -106,22 +110,28 @@ public void saveArchiveEntriesToVariables(DataWrapper archiveData, List<ArchiveV
     public void verifyArchiveContainsEntries(DataWrapper archiveData, List<NamedEntry> parameters)
     {
         Set<String> entryNames = ZipUtils.readZipEntryNamesFromBytes(archiveData.getBytes());
-
-        parameters.forEach(entry ->
+        boolean verificationFailed = false;
+        for (NamedEntry entry : parameters)
         {
             String expectedName = entry.getName();
             if (entry.getRule() != null)
             {
                 StringComparisonRule comparisonRule = entry.getRule();
-                softAssert.assertThat(String.format(
-                        "The archive contains entry matching the comparison rule '%s' with name pattern '%s'",
-                        comparisonRule, expectedName), entryNames, hasItem(comparisonRule.createMatcher(expectedName)));
+                verificationFailed = !softAssert.assertThat(String.format(
+                        "The archive contains entry matching the comparison "
+                        + "rule '%s' with name pattern '%s'", comparisonRule, expectedName), entryNames,
+                        hasItem(comparisonRule.createMatcher(expectedName))) || verificationFailed;
             }
             else
             {
-                softAssert.assertThat("The archive contains entry with name " + expectedName, entryNames,
-                        hasItem(expectedName));
+                verificationFailed = !softAssert.assertThat("The archive contains entry with name "
+                        + expectedName, entryNames, hasItem(expectedName)) || verificationFailed;
             }
-        });
+        }
+        if (verificationFailed)
+        {
+            attachmentPublisher.publishAttachment("archive-entries-result-table.ftl",
+                    Map.of("entryNames", entryNames), "Archive entries");
+        }
     }
```

Comment:
now it can be refactored in even more pretty way:
```java
        boolean verificationPassed = parameters.stream()
                .map(entry -> assertArchiveEntry(entry, entryNames))
                .reduce(true, (a, b) -> a && b);
        if (!verificationPassed)
        {
            attachmentPublisher.publishAttachment("archive-entries-result-table.ftl",
                    Map.of("entryNames", entryNames), "Archive entries");
        }
    }

    private boolean assertArchiveEntry(NamedEntry expectedEntry, Set<String> entryNames)
    {
        String expectedName = expectedEntry.getName();
        StringComparisonRule comparisonRule = expectedEntry.getRule();
        if (comparisonRule != null)
        {
            return softAssert.assertThat(String.format(
                    "The archive contains entry matching the comparison " + "rule '%s' with name pattern '%s'",
                    comparisonRule, expectedName), entryNames, hasItem(comparisonRule.createMatcher(expectedName)));
        }
        return softAssert.assertThat("The archive contains entry with name " + expectedName, entryNames,
                hasItem(expectedName));
    }
```

Reply:
thank you, done